react/CategoryPanel.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component, Fragment } from 'react'
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+import { Link, withRuntimeContext } from 'vtex.render-runtime'
+import { facetOptionShape } from './constants/propTypes'
+import searchResult from './searchResult.css'
+import CategoriesHighlights from 'vtex.store-components/CategoriesHighlights'
+
+class CategoryPanel extends Component {
+  static propTypes = {
+    /** Category tree */
+    tree: PropTypes.arrayOf(facetOptionShape),
+    /** Number of category filters to be shown */
+    noOfCategories: PropTypes.number,
+    /** Number of categories per row */
+    quantityOfItemsPerRow: PropTypes.number,
+  }
+
+  static defaultProps = {
+    tree: [],
+    noOfCategories: 1000,
+    quantityOfItemsPerRow: 4,
+  }
+
+  renderCategoryShelf = (
+    category,
+    noOfChildren,
+    quantityOfItemsPerRow,
+    mobile = false
+  ) => {
+    const categoriesHighlighted = category.Children.slice(0, noOfChildren).map(
+      child => ({
+        id: child.Id,
+        name: child.Name,
+        to: child.Link,
+      })
+    )
+    const headerClasses = classNames(
+      searchResult.categoryPanelHeaderRow,
+      'flex flex-row flex-wrap items-stretch pa3 bn',
+      {
+        mh4: !mobile,
```

Comment:
I think you could just use `mh4-ns` tachyon instead of a conditional with the mobile from `runtime` 🤔

Reply:
Good pick! Thanks! :sunglasses: